jenkins/opensearch-dashboards/integ-test.jenkinsfile

```diff
@@ -0,0 +1,137 @@
+lib = library(identifier: "jenkins@20211118", retriever: legacySCM(scm))
+
+pipeline {
+    options {
+        timeout(time: 3, unit: 'HOURS')
+    }
+    agent none
+    environment {
+        BUILD_MANIFEST = "build-manifest.yml"
+        BUILD_JOB_NAME = 'distribution-build-opensearch-dashboards'
+    }
+    parameters {
+        string(
+            name: 'TEST_MANIFEST',
+            description: 'Test manifest under the manifests folder, e.g. manifests/2.0.0/opensearch-dashboards-2.0.0-test.yml.',
+            trim: true
+        )
+        string(
+            name: 'BUILD_MANIFEST_URL',
+            description: 'The build manifest URL, e.g. https://ci.opensearch.org/ci/dbc/distribution-build-opensearch-dashboards/2.0.0/98/linux/x64/builds/opensearch-dashboards/manifest.yml.',
+            trim: true
+        )
+        string(
+            name: 'AGENT_LABEL',
+            description: 'The agent label where the tests should be executed, e.g. Jenkins-Agent-al2-x64-c54xlarge-Docker-Host.',
+            trim: true
+        )
+        string(
+            name: 'AGENT_IMAGE',
+            description: 'The agent label where the tests should be executed, e.g. opensearchstaging/ci-runner:centos7-x64-arm64-jdkmulti-node10.24.1-cypress6.9.1-20211028.',
+            trim: true
+        )
+    }
+    stages {
+        stage('verify-parameters') {
+            steps {
+                script {
+                    if (AGENT_LABEL == '') {
+                        currentBuild.result = 'ABORTED'
+                        error("Integration Tests failed to start. Missing parameter: AGENT_LABEL.")
+                    }
+                    if (AGENT_IMAGE == '') {
+                        currentBuild.result = 'ABORTED'
+                        error("Integration Tests failed to start. Missing parameter: AGENT_IMAGE.")
+                    }
+                }
+            }
+        }
+        stage('download-manifest') {
+            agent {
+                node {
```

Comment:
Should this be in a docker?

Reply:
Within OpenSearch integ-test, they are able to have a single step that downloads and runs the test. However, the image that is required to run the integration tests for OpenSearch Dashboards does not support `wget` which is what the `downloadBuildManifest` requires. 

Would the suggestion be to hardcode this stage to utilize a docker image that would normally be used by OpenSearch?

Also for my own curiosity, what's the benefit of running within a docker container vs running within a node? Does this prevent potential conflicts?

Reply:
Running within container ensures a fresh run every-time because the container will be destroyed after every run. Also, as you rightly mentioned, it also helps prevent conflicts. It would also ensure that the node is not affected by the job.

We can specify any image that would help support this operation. 

Also, why are we taking image name as input from the user? Can these possibly change without changes to the job?

Reply:
> We can specify any image that would help support this operation.

Do you know of which I can use? Used a couple docker images and all fail with `line 1: wget: command not found`. It would appear that's why OpenSearch is doing the same for integ tests: https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/opensearch/integ-test.jenkinsfile#L37

> Also, why are we taking image name as input from the user? 

When OpenSearch Dashboards builds it already detected the docker image and will require to use the appropriate image for example: https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.0.0/opensearch-dashboards-2.0.0.yml#L8. So when the build pipeline triggers the job it will pass the values it already has on hand.

> Can these possibly change without changes to the job?

Apologies I'm not clear what you mean here could you clarify?

Reply:
> When OpenSearch Dashboards builds it already detected the docker image and will require to use the appropriate image for example: https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.0.0/opensearch-dashboards-2.0.0.yml#L8. So when the build pipeline triggers the job it will pass the values it already has on hand.

Does this job pick the docker image from the manifest file? If yes, we should not have an input param,  but assign the image variable locally using manifest.

Reply:
> You can `curl` instead of `wget` I believe ...

Secret sauce, was able to update.

Reply:
> Does this job pick the docker image from the manifest file? If yes, we should not have an input param, but assign the image variable locally using manifest.

As of right now, the actual build manifest doesn't include the CI image stuff. The `manifest` do. There could be potential to pass `MANIFEST`, `TEST_MANIFEST`, and `BUILD_MANIFEST_URL`. But if we are going that route I would rather we open an issue and address that in both OpenSearch and OpenSearch Dashboards.

Reply:
@kavilla - In that case, can we code in the docker image rather than taking it as an input parameter? It would reduce one parameter for the user. 
Once the manifest supports holding the image, the job would then parse the build manifest and get the image from there

Reply:
We could utilize https://github.com/opensearch-project/opensearch-build/issues/1687, which would have the image available. But that wouldn't resolve the issue about the number of params. I wouldn't see any issue if we did include the CI information into the actual build manifest to reduce the amount of params.

I created an issue for that: https://github.com/opensearch-project/opensearch-build/issues/1694 since that will require more refactoring within the actual build repo.